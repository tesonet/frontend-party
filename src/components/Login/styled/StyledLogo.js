import React from 'react';
import styled from 'styled-components';

import LogoImage from '../../../assets/logo.png';
import { login as copy } from '../../../assets/copy/global.json';

const StyledContainer = styled.div`
  margin-bottom: 4rem;
  text-align: center;

  img {
    max-width: 70%;
  }
`;

const StyledLogo = props => (
  <StyledContainer>
    <img src={LogoImage} alt={copy.logoAltText} {...props} />
  </StyledContainer>
);

export default StyledLogo;
