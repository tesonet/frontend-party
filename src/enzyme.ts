import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
