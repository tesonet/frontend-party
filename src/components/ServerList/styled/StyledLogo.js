import React from 'react';
import styled from 'styled-components';

import LogoImage from '../../../assets/logo-dark.png';

const StyledContainer = styled.img`
  max-width: 6rem;
`;

const StyledLogo = props => (
  <StyledContainer src={LogoImage} alt="Testio logo" {...props} />
);

export default StyledLogo;
