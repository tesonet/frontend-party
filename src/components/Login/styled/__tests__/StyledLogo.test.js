import React from 'react';
import { shallow } from 'enzyme';

import StyledLogo from '../StyledLogo';

describe('StyledLogo component', () => {
  it('should render the <div /> element.', () => {
    const renderedComponent = shallow(<StyledLogo />).dive();

    expect(renderedComponent.type()).toEqual('div');
    expect(renderedComponent).toMatchSnapshot();
  });
});
