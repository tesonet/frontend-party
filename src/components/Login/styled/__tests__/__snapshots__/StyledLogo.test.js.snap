// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLogo component should render the <div /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div>
    <img
        alt="Testio logo"
        src="logo.png"
    />
</styled.div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Testio logo"
        src="logo.png"
/>,
      "className": "sc-bdVaJa fjsemD",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Testio logo",
        "src": "logo.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Testio logo"
          src="logo.png"
/>,
        "className": "sc-bdVaJa fjsemD",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Testio logo",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
